src/utils.jl

```diff
@@ -54,3 +54,53 @@ function checkbalanced(x::AbstractVecOrMat, y::AbstractVecOrMat)
         throw(ArgumentError("source and target marginals are not balanced"))
     return nothing
 end
+
+struct FiniteDiscreteMeasure{X<:AbstractArray,P<:AbstractVector}
+    support::X
+    p::P
+
+    function FiniteDiscreteMeasure{X,P}(support::X, p::P) where {X,P}
+        size(support, 1) == length(p) ||
+            error("number of rows of `support` and `p` must be equal")
+        sum(p) ≈ 1 || error("`p` must sum to 1")
+        all(p .>= 0) || error("values of `p` must be greater of equal than 0")
+        return new{X,P}(support, p)
+    end
+end
+
+"""
+    FiniteDiscreteMeasure(support::AbstractArray, p::AbstractVector)
+Construct a finite discrete probability measure with support `support` and corresponding weights `p`.
+"""
+function FiniteDiscreteMeasure(support::AbstractArray, p::AbstractVector)
+    P = sum(p)
+    if size(support, 2) == 1
+        return if P ≈ 1
+            DiscreteNonParametric(vec(support), p)
+        else
+            DiscreteNonParametric(vec(support), p ./ P)
+        end
+    else
+        return if P ≈ 1
+            FiniteDiscreteMeasure{typeof(support),typeof(p)}(support, p)
+        else
+            FiniteDiscreteMeasure{typeof(support),typeof(p)}(support, p ./ P)
+        end
+    end
+end
```

Comment:
Yeah, the sorting has to be clearly stated, I got tricked by it sometimes. About the normalization, it is a matter of convenience. It allows one to write `mu = FiniteDiscreteMeasure(rand(10,2), rand(10))`, which I find really neat. Of course, if it creates significant issues, then we should just drop it. Otherwise, I like how it allows one to write very clean code.